File: LeetCode/Python/jumpGame.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        dp = [False for x in range(len(nums))]
        dp[0] = True

        for j in range(1, len(nums)):
            for i in range(0, j):
                if dp[i] and i + nums[i] >= j:
                    dp[j] = True
                    break
        return dp[len(dp) - 1]

File: LeetCode/CSharp/SecondRound/JumpGame.cs
namespace LeetCode.CSharp.SecondRound
{
    public class JumpGame
    {
        public bool CanJump(int[] nums)
        {
            int maxIndex = 0;
            int n = nums.Length;

            for (int i = 0; i < n; i++)
            {
                if (i > maxIndex)
                    return false;
                else
                    maxIndex = Math.Max(maxIndex, i + nums[i]);
            }

            return maxIndex >= n - 1;
        }
    }
}